### PR TITLE
Update logic for detecting if the output stream is connected to a terminal

### PR DESCRIPTION
**Pull request by cushon**
https://github.com/remkop/picocli/issues/2083

### CODE REVIEW

File: src/main/java/picocli/CommandLine.java [hunk 1]
```diff
@@ -17840,7 +17840,18 @@ static final boolean forceEnabled() { return System.getenv("CLICOLOR_FORCE") !=
...
             /** http://stackoverflow.com/questions/1403772/how-can-i-check-if-a-java-programs-input-output-streams-are-connected-to-a-term */
             static boolean calcTTY() {
-                try { return System.class.getDeclaredMethod("console").invoke(null) != null; }
+                try {
+                    Object console = System.class.getDeclaredMethod("console").invoke(null);
+                    if (console == null) {
+                        return false;
+                    }
+                    try {
+                        Method isTerminal = Console.class.getDeclaredMethod("isTerminal");
```
Comment (remkop) [inline src/main/java/picocli/CommandLine.java:17849]: Looks good, but the `Console` class was introduced in Java 6, and picocli supports Java 5, so can you change this to find the `Console` class via reflection?

**Review by remkop — CHANGES_REQUESTED** [review]

**Comment by remkop** [comment]
Thank you for raising this! Overall the PR looks good, please look at the minor change I requested to find the `Console` class via reflection.

File: src/main/java/picocli/CommandLine.java (hunk 1 above)
Comment (cushon) [reply src/main/java/picocli/CommandLine.java:17849]: Thanks, done

**Review by remkop — APPROVED** [review]

**Comment by remkop** [comment]
Merged. Thank you for the contribution!

**Pull request merged**